api/src/main/java/jakarta/data/metamodel/Attribute.java

```diff
@@ -25,111 +25,93 @@
-     * @throws MappingException if an entity attribute with this name does not exist or
-     *                          if no Jakarta Data provider provides a repository for the entity type.
-     */
-    public final Sort desc() {
-        return attrInfo().desc();
-    }
-
-    /**
-     * Obtain a request for a descending, case insensitive {@link Sort} based on the entity attribute.
+     * <p>The default value of <code>false</code> is appropriate for instances that are
+     * pre-initialized, such as those that are provided by a Jakarta Data provider
+     * or are generated in advance with an annotation processor.</p>
      *
-     * @return a request for a descending, case insensitive sort on the entity attribute.
-     * @throws MappingException if an entity attribute with this name does not exist or
-     *                          if no Jakarta Data provider provides a repository for the entity type.
+     * @param attr attribute implementation provided by the Jakarta Data provider.
+     * @return true if initialization was successful; false if already initialized.
      */
-    public final Sort descIgnoreCase() {
-        return attrInfo().descIgnoreCase();
+    default boolean init(Attribute attr) {
+        return false;
     }
 
     /**
      * <p>Obtains a new instance for the Jakarta Data provider to initialize.</p>
      *
      * <p>The Jakarta Data provider automatically initializes the instance when
      * used as the value of a {@code public}, {@code static}, {@code final} field
-     * of a class that is annotated with {@link StaticMetamodel},
+     * of a class or interface that is annotated with {@link StaticMetamodel},
      * where the field name matches the name of an entity attribute.
-     * For example, a {@code Vehicle} entity with attributes {@code vin}, {@code make},
-     * {@code model}, and {@code year} could have the following static metamodel,</p>
+     * For example, a {@code Person} entity with attributes {@code id}, {@code emails},
+     * {@code phoneNumbers}, and {@code name} could have the following static metamodel,</p>
      *
      * <pre>
-     * &#64;StaticMetamodel(Vehicle.class)
-     * public class Vehicle_ {
-     *     public static final Attribute vin = Attribute.get();
-     *     public static final Attribute make = Attribute.get();
-     *     public static final Attribute model = Attribute.get();
-     *     public static final Attribute year = Attribute.get();
+     * &#64;StaticMetamodel(Person.class)
+     * public interface Person {
+     *     SortableAttribute id = SortableAttribute.get();
+     *     Attribute emails = Attribute.get();
+     *     Attribute phoneNumbers = Attribute.get();
+     *     TextAttribute name = TextAttribute.get();
      * }
      * </pre>
      *
      * @return a new instance for an entity attribute.
      */
-    public static final Attribute get() {
-        return new Attribute();
-    }
+    public static Attribute get() {
```

Comment:
This method does not make sense either.
Let the vendor do it their way.

Reply:
> This method does not make sense either. Let the vendor do it their way.

This method is essential for allowing the user to initialize the metamodel field for the case where the metamodel is initialized at run time,

```public static final Attribute price = Attribute.get();```

By telling me to get rid of it and let the vendor do it means that build-time will be the only approach possible for the static metamodel, and that is completely unacceptable.

Also, it should be pointed out that we already had a `public static final Attribute get()` method prior to this pull, and I've only removed the `final` from it because `final` is implicit after that class was switched to an interface.

Reply:
But why not set a field instance?

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/reflect/Field.html#set(java.lang.Object,java.lang.Object)

Reply:
> But why not set a field instance?
> 
> https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/reflect/Field.html#set(java.lang.Object,java.lang.Object)

Java does not allow setting a field instance when the field is `final`.  You also cannot leave a `final` value unset and set it for the first time later.  Java `final` doesn't allow it.   If you want to set it later, you need non-final, and you become vulnerable to being overwritten and contaminated by any code that decides to change it, as was the concern in #296 .  Again, this pull is the wrong place for that discussion.  Go back and reopen #296 if you want to have that discussion.  The requirement for`final` is already present and we should not abuse this pull to change that.